x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts

```diff
@@ -21,32 +26,51 @@ export async function getCategories(options: Registry.CategoriesParams) {
   return Registry.fetchCategories(options);
 }
 
+// TODO this function is odd to me, it will use `fetchList` to get the **latest** packages and then squash them together
+// with the installed saved objects which could be on a completely separate version.
+// This causes issues for the `getPackageKeysByStatus` function because it is used in places that assumes it will
+// retrieve the `installed` information for a package and instead it gets the latest information
 export async function getPackages(
   options: {
     savedObjectsClient: SavedObjectsClientContract;
   } & Registry.SearchParams
 ) {
   const { savedObjectsClient, experimental, category } = options;
-  const registryItems = await Registry.fetchList({ category, experimental }).then((items) => {
+  // todo remove the .then and see if that fixes the unhandled promise rejection
+  /* const registryItems = await Registry.fetchList({ category, experimental }).then((items) => {
```

Comment:
@jfsiii if this gets a `RegistryError` thrown would it cause an unhandled promise rejection?